src/main/java/org/generationcp/middleware/dao/GermplasmDAO.java

```diff
@@ -1710,4 +1716,38 @@ public boolean hasExistingCrosses(final Integer femaleParent, final List<Integer
 		}
 	}
 
+	public GermplasmDto getGermplasmDtoByGid(final Integer gid) {
+		final StringBuilder queryBuilder =
+			new StringBuilder(FIND_GERMPLASM_MATCHES_MAIN_QUERY);
```

Comment:
I think we can reduce duplicated here if we just add a match by GID in `GermplasmMatchRequestDto` 
`private List<String> gids;`
then expand `addGermplasmMatchesFilter` to filter by gids. Then getGermplasmDtoByGid here can just call `findGermplasmMatches` returning the first DTO matched by GID (if any)

Reply:
@darla-leafnode  I wont mess with the matches query, It was complicated to find an index that works well by the time I implemented it. That's the main reason I have created another query, so I am sure I dont affect the matches performance that it is used in the new import 
CC @abatac